optlang/glpk_interface.py

```diff
@@ -579,6 +579,9 @@ def __setstate__(self, repr_dict):
         if repr_dict['glpk_status'] == 'optimal':
             self.optimize()  # since the start is an optimal solution, nothing will happen here
 
```

Comment:
The problem is that `__del__` is only being called when the object is garbage collected (which it isn't in this case if it has a `__del__` method) :(